packages/fcl/src/exec/verify.js

```diff
@@ -66,25 +58,53 @@ const VERIFY_SIG_SCRIPT = `
   }
       
   pub fun main(
+    acctAddress: Address,
     message: String,
-    rawPublicKeys: [String],
-    weights: [UFix64],
-    signAlgos: [UInt],
-    hashAlgos: [UInt],
+    keyIds: [Int],
     signatures: [String],
   ): Bool {
 
     let keyList = Crypto.KeyList()
+    let rawPublicKeys: [String] = []
+    let weights: [UFix64] = []
+    let signAlgos: [UInt] = []
+    let hashAlgos: [UInt] = []
+    let uniqueKeys: {Int: Bool} = {}
+    let account = getAccount(acctAddress)
     
+    for id in keyIds {
+      uniqueKeys[id] = true
+    }
+
+    assert(uniqueKeys.keys.length == keyIds.length, message: "Invalid KeyId: Duplicate key found for account")
+
+    var counter = 0
+    while (counter < keyIds.length) {
+      let accountKey = account.keys.get(keyIndex: keyIds[counter]) ?? panic("Key provided does not exist on account")
+      rawPublicKeys.append(String.encodeHex(accountKey.publicKey.publicKey))
+      weights.append(accountKey.weight)
+      signAlgos.append(UInt(accountKey.publicKey.signatureAlgorithm.rawValue))
+      hashAlgos.append(UInt(accountKey.hashAlgorithm.rawValue))
+      counter = counter + 1
+    }
+
+    var totalWeight = 0.0
+    var weightIndex = 0
+    while (weightIndex < weights.length) {
+      totalWeight = totalWeight + weights[weightIndex]
+      weightIndex = weightIndex + 1
+    }
+    assert(totalWeight >= 1000.0, message: "Signature key weights do not meet threshold >= 1000.0")
+
     var i = 0
     for rawPublicKey in rawPublicKeys {
       keyList.add(
         PublicKey(
           publicKey: rawPublicKey.decodeHex(),
-          signatureAlgorithm: signAlgos[i] == 2 ? SignatureAlgorithm.ECDSA_P256 : SignatureAlgorithm.ECDSA_secp256k1 
+          signatureAlgorithm: signAlgos[i] == 2 ? SignatureAlgorithm.ECDSA_secp256k1  : SignatureAlgorithm.ECDSA_P256
```

Comment:
Do we need to be concerned about `BLS_BLS12_381` ? https://docs.onflow.org/cadence/language/crypto/#signing-algorithms

Reply:
Don't believe so as it's not supported for signatures. @tarakby can confirm I think.

Reply:
Yep, BLS is supported by Cadence but not supported by account signatures.